db/migrate/20171011111414_create_users.rb

```diff
@@ -0,0 +1,9 @@
+class CreateUsers < ActiveRecord::Migration[5.1]
+  def change
+    create_table :users do |t|
+      t.string :username, :null => false, :unique => true
+      t.string :password_hash, :null => false, :length => {minimum: 6 }
```

Comment:
Heads up: in the database, this is a `BCrypt` hash, which is always 60 characters in length if my memory is correct. Thus you will want to validate the length of the input from the user, not the string stored in the database.